tests/nnstreamer_source/unittest_src_iio.cc

```diff
@@ -1564,6 +1564,85 @@ TEST (test_tensor_src_iio, data_verify_freq_generic_type)
   clean_iio_dev_structure (dev0);
 }
 
+/**
+ * @brief test the logic with invalid frequency value
+ */
+TEST (test_tensor_src_iio, set_frequency_n)
+{
+  iio_dev_dir_struct *dev0;
+  GstElement *src_iio_pipeline;
+  GstElement *src_iio;
+  GstStateChangeReturn status;
+  gchar *parse_launch;
+  gint data_value;
+  guint data_bits;
+  gint ret_freq;
+
+  data_value = DATA;
+  data_bits = 16;
+  /** Make device */
+  dev0 = make_full_device (data_value, data_bits);
+  ASSERT_NE (dev0, nullptr);
+  /** setup */
+  dev0->log_file = g_build_filename (dev0->base_dir, "temp.log", NULL);
+  parse_launch =
+      g_strdup_printf
+      ("%s iio-base-dir=%s device-number=%d name=my-src-iio ! fakesink",
+      ELEMENT_NAME, dev0->iio_base_dir_sim, 0);
+  src_iio_pipeline = gst_parse_launch (parse_launch, NULL);
+  g_free (parse_launch);
+
+  /** get and verify the caps */
+  src_iio = gst_bin_get_by_name (GST_BIN (src_iio_pipeline), "my-src-iio");
```

Comment:
parse_launch might be null when gst_parse_lunch fails. https://gstreamer.freedesktop.org/documentation/gstreamer/gstparse.html?gi-language=c#gst_parse_launch


Note that "you might get a return value that is not NULL even though the error is set. In this case there was a recoverable parsing error and you can try to play the pipeline."

Reply:
`nullptr` checker has been added before calling `g_free`

Reply:
You need to check ```src_iio_pipeline```, not ```parse_launch``` at LINE 1593.

Reply:
@myungjoo it's fixed